lib/OpenQA/Schema/Result/ScheduledProducts.pm

```diff
@@ -275,6 +273,19 @@ sub _schedule_iso {
             my $prio = delete $settings->{PRIO};
             $settings->{_GROUP_ID} = delete $settings->{GROUP_ID};
 
+            # Any setting name ending in _URL is special: it tells us to download
+            # the file at that URL before running the job
+            my $download = create_downloads_list($settings);
+            foreach my $url (keys %$download) {
+                my $value = $download->{$url};
+                unless ($tmp_downloads{$url}) {
+                    $tmp_downloads{$url} = {destination => {$value->[0] => 1}, do_extract => $value->[1]};
+                    next;
+                }
+                $tmp_downloads{$url}->{destination}->{$value->[0]} = 1
+                  unless ($tmp_downloads{$url}->{destination}->{$value->[0]});
```

Comment:
This would only take the `do_extract` parameter from the first download we encounter into account. Handling the extraction parameter per destination would likely be the correct thing to do. However, that would complicate things further, e.g. there might be the case when some downloads should be extracted and some not.

You could at least add a comment about the problem, e.g. `# caveat: The extraction parameter is currently not processed per destination. If multiple destinations for the same download have a different 'do_extract' parameter the first one will win.`.

Reply:
Thanks for pointing this issue. I did not realize it. So should we keep this logic and just add the comments you mentioned?

Reply:
For this commit I'd definitely just add the caveat comment. You might want to actually fix this in a 2nd commit but I'm also not sure whether it is worth it. Hence I wouldn't require a fix for this PR.

Reply:
I do not want to fix this :)